### PR TITLE
Fix Specs. Fix Rubocop. Update addressable gem.

### PR DESCRIPTION
**Pull request by anovadox**
The specs were using out of date `should` matchers for rspec, and also making inefficient network calls.  They were also all failing..   I modernized the rspec side and introduced VCR to make the specs faster and more reliable. I cleaned up all of the lingering rubocop violations and fixed the tests. (Making small changes to conform to current diffbot api documentation)

And I updated the addressable gem version.
https://www.pivotaltracker.com/story/show/161666605

### CODE REVIEW

File: spec/lib/supported/analyze_spec.rb [hunk 1]
```diff
@@ -2,31 +2,32 @@
 
 # more tests to come...
 describe Biffbot::Analyze do
-  before :each do
-    url = 'http://www.huffingtonpost.ca/2012/10/11/amanda-todd-teen-bullying-suicide-youtube_n_1959668.html'
-    Biffbot.token = ENV['DIFFBOT_DEV_TOKEN']
-    @article = Biffbot::Analyze.new(url, type: 'article')
+  before :all do
+    VCR.use_cassette('analyze') do
+      url = 'https://theintercept.com/2018/10/31/midterm-elections-2018-rural-progressives/'
```
Comment (TheMetalCode) [inline spec/lib/supported/analyze_spec.rb]: Love me some Intercept

File: biffbot.gemspec [hunk 2]
```diff
@@ -13,17 +15,19 @@ Gem::Specification.new do |gem|
...
+  gem.add_development_dependency('activesupport')
+  gem.add_development_dependency('dotenv')
   gem.add_development_dependency('rake')
+  gem.add_development_dependency('rspec')
   gem.add_development_dependency('rubocop')
-  gem.add_development_dependency('dotenv')
+  gem.add_development_dependency('vcr', '~> 4.0')
+  gem.add_development_dependency('webmock', '~> 3.4')
   gem.add_development_dependency('yard')
-  gem.add_development_dependency('codeclimate-test-reporter')
```
Comment (benjaminoakes) [inline biffbot.gemspec]: This was removed, which seems appropriate, but is perhaps hard to notice.  Are there any other changes that weren't automatic?

File: lib/biffbot/bulk.rb [hunk 3]
```diff
@@ -24,11 +24,12 @@ def create_job name, api_type, urls = [], options = {}
...
       api_url = parse_options(options, api_url).display_uri.to_s
-      endpoint = 'http://api.diffbot.com/v3/bulk'
-      post_body = generate_post_body(name, api_url, urls, options)
-      www_form_encoded_body = URI.encode_www_form(post_body)
 
-      JSON.parse(HTTParty.post(endpoint, body: www_form_encoded_body, headers: {'Content-Type' => 'application/x-www-form-urlencoded'}).body).each_pair do |k, v|
+      JSON.parse(
+        HTTParty.post('http://api.diffbot.com/v3/bulk',
+                      body: URI.encode_www_form(generate_post_body(name, api_url, urls, options)),
+                      headers: {'Content-Type' => 'application/x-www-form-urlencoded'}).body
```
Comment (benjaminoakes) [inline lib/biffbot/bulk.rb]: Why single quotes?  (Are we using the same style as on other projects?)

File: lib/biffbot/base.rb [hunk 4]
```diff
@@ -83,5 +76,21 @@ def parse_options options = {}, request = ''
 
       uri
     end
+
+    private
+
+    # handle different option types
+    #
+    # @param options [Hash] An hash of options
```
[the remark this answers is not in the document]
Comment (anovadox) [reply lib/biffbot/base.rb]: Following the above conventions, happy to remove

File: lib/biffbot/bulk.rb (hunk 3 above)
Comment (anovadox) [reply lib/biffbot/bulk.rb]: The style-update PR hadn't been merged.  I'm sure we can get this fixed in a second PR if necessary (but this is a fork of a public gem)

File: spec/lib/supported/analyze_spec.rb (hunk 1 above)
Comment (benjaminoakes) [reply spec/lib/supported/analyze_spec.rb]: Why did this URL change?

File: spec/lib/supported/article_spec.rb [hunk 5]
```diff
@@ -2,26 +2,27 @@
 
 # currently tests v2 article api
 describe Biffbot::Article do
-  before :each do
-    url = 'http://www.huffingtonpost.ca/2012/10/11/amanda-todd-teen-bullying-suicide-youtube_n_1959668.html'
-    Biffbot.token = ENV['DIFFBOT_DEV_TOKEN']
-    @article = Biffbot::Article.new(url)
+  before :all do
+    VCR.use_cassette('article') do
+      url = 'https://theintercept.com/2018/10/31/midterm-elections-2018-rural-progressives/'
```
Comment (benjaminoakes) [inline spec/lib/supported/article_spec.rb]: Why did this URL change?

File: spec/lib/supported/image_spec.rb [hunk 6]
```diff
@@ -2,26 +2,27 @@
 
 # currently tests v2 image api
 describe Biffbot::Image do
-  before :each do
-    url = 'http://www.huffingtonpost.ca/2012/10/11/amanda-todd-teen-bullying-suicide-youtube_n_1959668.html'
-    Biffbot.token = ENV['DIFFBOT_DEV_TOKEN']
-    @image = Biffbot::Image.new(url)
+  before :all do
+    VCR.use_cassette('image') do
+      url = 'https://theintercept.com/2018/10/31/midterm-elections-2018-rural-progressives/'
```
Comment (benjaminoakes) [inline spec/lib/supported/image_spec.rb]: Why did this URL change?

This is not an image, so it's explicitly incorrect.

File: biffbot.gemspec (hunk 2 above)
Comment (anovadox) [reply biffbot.gemspec]: I corrected https://github.com/doximity/biffbot/pull/3/files#diff-52c976fc38ed2b4e3b1192f8a8e24cffR11
(a copy-paste typo from the original rakefile)
I also needed to import activesupport `blank` 
and I updated the addressable gem 😄

File: spec/lib/supported/product_spec.rb [hunk 7]
```diff
@@ -2,31 +2,32 @@
 
 # currently tests v2 product api
 describe Biffbot::Product do
-  before :each do
-    url = 'http://www.amazon.com/Google-Chromecast-Streaming-Media-Player/dp/B00DR0PDNE/'
-    Biffbot.token = ENV['DIFFBOT_DEV_TOKEN']
-    @product = Biffbot::Product.new(url)
+  before :all do
+    VCR.use_cassette('product') do
+      url = 'https://www.amazon.com/Amazon-Kindle-eReader-6-Inch-Touchscreen/dp/B00ZV9PXP2/'
```
Comment (benjaminoakes) [inline spec/lib/supported/product_spec.rb]: Why did this URL change?

**Pull request closed without merging**